src/parameterizations/lateral/MOM_wave_drag.F90

```diff
@@ -107,15 +135,39 @@ subroutine wave_drag_calc(u, v, drag_u, drag_v, G, CS)
 
   Drag_u(:,:) = 0.0 ; Drag_v(:,:) = 0.0
 
-  !$OMP do
-  do k=1,CS%nf ; do j=js,je ; do I=is-1,ie
-    Drag_u(I,j) = Drag_u(I,j) + u(I,j,k) * CS%coef_u(I,j,k)
-  enddo ; enddo ; enddo
-
-  !$OMP do
-  do k=1,CS%nf ; do J=js-1,je ; do i=is,ie
-    Drag_v(i,J) = Drag_v(i,J) + v(i,J,k) * CS%coef_v(i,J,k)
-  enddo ; enddo ; enddo
+  if (CS%tensor_drag) then
+    !$OMP do
+    do k=1,CS%nf ; do j=js,je
+      do I=is,ie-1
+        Drag_u(I,j) = Drag_u(I,j) + u(I,j,k) * CS%coef_u(I,j,k) + &
+                      .25 * (v(i+1,J,k) + v(i,J-1,k) + v(i,J,k) + v(i+1,J-1,k)) * CS%coef_uv(I,j,k)
```

Comment:
This expression lacks the parentheses that would be needed to respect rotational symmetry.